CMakeLists.txt

```diff
@@ -52,36 +48,32 @@ if(UNIX)
   message("SDL Include Dir is " ${SDL_INCLUDE_DIR})
 endif()
 
-if(UNIX AND NOT APPLE)
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   message("Building for Linux")
-  add_definitions( -D_DEBUG -D__LINUX__ -DLINUX -D_MAX_PATH=260 -D_MAX_FNAME=256 -D_REENRANT -D__32BIT__ -DHAVEALLOCA_H -D_USE_OGL_ACTIVE_TEXTURES)
+  add_compile_definitions(_DEBUG __LINUX__ LINUX _MAX_PATH=260 _MAX_FNAME=256 _REENRANT __32BIT__ HAVEALLOCA_H _USE_OGL_ACTIVE_TEXTURES)
   set(PLATFORM_INCLUDES "lib/linux" ${SDL_INCLUDE_DIR})
-endif()
-
-if(APPLE)
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
   message("Building for MAC OSX")
```

Comment:
I think these messages can be removed as well, it's not like someone is going to cross compile Linux <-> macOS or has forgotten what system they are on :)